src/views/catalog/CreateFromInstanceTypes/components/AddBootableVolumeModal/AddBootableVolumeModal.tsx

```diff
@@ -68,116 +42,25 @@ const AddBootableVolumeModal: FC<AddBootableVolumeModalProps> = ({
     initialBootableVolumeState,
   );
 
-  const {
-    bootableVolumeName,
-    size,
-    annotations,
-    labels,
-    pvcName,
-    pvcNamespace,
-    uploadFile,
-    storageClassName,
-  } = bootableVolume || {};
+  const { bootableVolumeName, labels } = bootableVolume || {};
 
   const isUploadForm = formSelection === RADIO_FORM_SELECTION.UPLOAD_IMAGE;
 
-  const instanceTypesToSizesMap = useMemo(
-    () => getInstanceTypesToSizesMap(instanceTypesNames.sort((a, b) => a.localeCompare(b))),
-    [instanceTypesNames],
-  );
-  const instanceTypeAndSize = useMemo(
-    () => labels?.[DEFAULT_INSTANCETYPE_LABEL]?.split('.') || ['', ''],
-    [labels],
-  );
-
-  const [instanceType, setInstanceType] = useState<string>(instanceTypeAndSize[0]);
-  const [instanceSize, setInstanceSize] = useState<string>(instanceTypeAndSize[1]);
-  const [sizeOptions, setSizeOptions] = useState<string[]>(
-    instanceTypesToSizesMap[instanceType] || [],
+  const setBootableVolumeField = useCallback(
+    (key: string, fieldKey?: string) => (value: string) =>
+      fieldKey
+        ? setBootableVolume((prevState) => ({
+            ...prevState,
+            [key]: { ...prevState[key], [fieldKey]: value },
+          }))
```

Comment:
check is this can work:
setBootableVolume((prevState) => ({
      ...prevState,
      ...(fieldKey ? { [key]: { ...prevState[key], [fieldKey]: value } } : { [key]: value }),
    }));

then u dont need line 56